server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/mdc/LoggingRequestFilter.java

```diff
@@ -0,0 +1,45 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.mdc;
+
+import java.io.Closeable;
+
+import org.apache.james.util.MDCBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+
+public class LoggingRequestFilter implements Filter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingRequestFilter.class);
+    private static final String BODY = "body";
+
+    @Override
+    public void handle(Request request, Response response) throws Exception {
+        try (Closeable closeable =
+                 MDCBuilder.create()
```

Comment:
You trick the MDCContext to do structured logging.
We should have an API for that so that we express our intent better

Reply:
Maybe, but it do not seem easy to do...

Reply:
And you only do easy things ?

Reply:
I'm lacking other examples where we would value structured logging.

I'm searching other solutions to get structured logging out of the box with our logback -> Kibana plugin and can not figure it out.

Finally, I don't want to invest time on a ticket that, technically, is not related to a Gitlab issue.

Reply:
I don't understand your arguments. You do structured logging. But you do it without making it explicit. So either we make it explicit or we don't do it at all.

Reply:
Do you mean "provide explicit API for structured loging that happens to use MDC"

Like:

```
StructuredLogging.forLogger(LOGGER)
    .addField(name, value)
    .log(logger -> logger.info("message"));
```